Tests/GeoLib/TestOctTree.cpp

```diff
@@ -0,0 +1,325 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include "gtest/gtest.h"
+#include <ctime>
+#include <random>
+
+#include "GeoLib/OctTree.h"
+#include "GeoLib/AABB.h"
+#include "GeoLib/Point.h"
+
+class GeoLibOctTree : public testing::Test
+{
+public:
+	typedef std::vector<GeoLib::Point*> VectorOfPoints;
+
+	GeoLibOctTree()
+	{}
+
+	void
+	checkOctTreeChildsNonNullptr(GeoLib::OctTree<GeoLib::Point> const& oct_tree) const
+	{
+#ifndef NDEBUG
+		ASSERT_NE(nullptr, oct_tree.getChild(0));
+		ASSERT_NE(nullptr, oct_tree.getChild(1));
+		ASSERT_NE(nullptr, oct_tree.getChild(2));
+		ASSERT_NE(nullptr, oct_tree.getChild(3));
+		ASSERT_NE(nullptr, oct_tree.getChild(4));
+		ASSERT_NE(nullptr, oct_tree.getChild(5));
+		ASSERT_NE(nullptr, oct_tree.getChild(6));
+		ASSERT_NE(nullptr, oct_tree.getChild(7));
+#endif
+	}
+
+	void
+	checkOctTreeChildsNullptr(GeoLib::OctTree<GeoLib::Point> const& oct_tree) const
+	{
+#ifndef NDEBUG
+		ASSERT_EQ(nullptr, oct_tree.getChild(0));
+		ASSERT_EQ(nullptr, oct_tree.getChild(1));
+		ASSERT_EQ(nullptr, oct_tree.getChild(2));
+		ASSERT_EQ(nullptr, oct_tree.getChild(3));
+		ASSERT_EQ(nullptr, oct_tree.getChild(4));
+		ASSERT_EQ(nullptr, oct_tree.getChild(5));
+		ASSERT_EQ(nullptr, oct_tree.getChild(6));
+		ASSERT_EQ(nullptr, oct_tree.getChild(7));
+#endif
+	}
+
+protected:
+	void
+	generateEquidistantPoints3d(std::size_t const n = 11)
+	{
+		for (std::size_t k(0); k<n; ++k) {
+			double const z(k-(n-1)/2.0);
+			for (std::size_t j(0); j<n; ++j) {
+				double const y(j-(n-1)/2.0);
+				for (std::size_t i(0); i<n; ++i) {
+					ps_ptr.push_back(new GeoLib::Point(i-(n-1)/2.0, y, z));
+				}
+			}
+		}
+	}
+
+protected:
+	VectorOfPoints ps_ptr;
+};
+
+TEST_F(GeoLibOctTree, TestWithEquidistantPoints3d)
+{
+	generateEquidistantPoints3d();
+	double const eps(10*std::numeric_limits<double>::epsilon());
+	GeoLib::OctTree<GeoLib::Point> oct_tree(
+		GeoLib::OctTree<GeoLib::Point>::createOctTree(*ps_ptr.front(),
+		*ps_ptr.back(), eps, 2));
+
+#ifndef NDEBUG
+	MathLib::Point3d const& ll(oct_tree.getLowerLeftCornerPoint());
```

Comment:
The reason I activated this code only in debug mode is that some additional methods in the OctTree are only activated in debug mode. These OctTree methods are useful only to investigate the structure of the OctTree object at hand and are not needed in the "usual" use. So a removal of `#ifndef NDEBUG` here needs a removal of `#ifndef NDEBUG` in OctTree. I am unsure if this is a good idea.